R/safe2x2Test.R

```diff
@@ -1,1148 +1,2016 @@
-#' designSafeTwoProportions(deltaMin = 0.7, sampleSizeRatio = 2, lowN = 30)
-#' #example with prior knowledge about point null hypothesis and minimum relative risk:
-#' res_precognition <- designSafeTwoProportions(stepSize = 100, priorInfo = list(h0 = 0.15, RR = 0.5))
-designSafeTwoProportions <- function(deltaMin = NA, alpha = 0.05, beta = 0.20,
-                                     alternative = c("two.sided", "greater", "less"), M = 100,
-                                     lowN = 20, highN = 200, sampleSizeRatio = 1,
-                                     lowDelta = 0.1, highDelta = 0.8, tol = 0.1, numberForSeed = NA,
-                                     priorInfo = NA, stepSize = 1) {
-
-  alternative <- match.arg(alternative)
-
-  #initialize starting parameters
-  deltas <- seq(lowDelta, highDelta, by = tol)
-
-  #calculate the number of complete groups a and b collected at low N
-  #for example lowN = 20, sampleSizeRatio = 1, then we start with 10 groups
-  #that have been collected
-  nGroups <- lowN/(sampleSizeRatio+1)
-  if (abs(round(nGroups) - nGroups) > 1e-8)
-    stop("lowN should be multiple of a.iter + b.iter")
-
-  #in the iteration we start directly with incrementing nGroups,
-  #so now subtract 1
-  nGroups <- nGroups - 1
-  nb <- sampleSizeRatio*(nGroups)
-  na <- nGroups
-  n <- na + nb
-
-  #for H1_delta.min, we want to test the power:
-  if(!any(is.na(priorInfo))){
-    #sampling from a point alternative hypothesis of interest
-    H1DeltaMin <- matrix(c(priorInfo[["h0"]], priorInfo[["RR"]]*priorInfo[["h0"]]),
-                         nrow = 1)
-    #do not need to try different deltas: one best S.
-    deltas <- 1
-  } else {
-    H1DeltaMin <- create_data_generating_distributions(deltaDesign = deltaMin, alternative = alternative,
-                                                       length.out = 5)
-  }
-
-  notreached <- TRUE
-
-  #save the start time to use as a seed inside the sampling process
-  #so the same set of samples from H1 delta.circ will be presented
-  #to each S-test constructed with delta.star
-
-  seedTime <- ifelse(is.na(numberForSeed), Sys.time(), numberForSeed)
-
-  cat("Trying n = ")
-
-  while (notreached & (n < highN)) {
-    #increment the group sizes
-    nGroups <- nGroups + stepSize
-    na <- nGroups
-    nb <- sampleSizeRatio*nGroups
-    if(nb %%1 != 0){
-      #no whole group
-      next
-    }
-
-    n <- na + nb
-    cat(paste0(n, " "))
-
-    #if na = nb and we are testing twosided we only need to test the delta ump,
-    #since this is then known analytically
-
-    if (na == nb & alternative == "two.sided" & any(is.na(priorInfo))) {
-      delta.ump <- get_delta_ump_analytically(n = na + nb, alpha = alpha,
-                                              lower = 0.1, upper = 0.9)
-
-      if (!is.na(delta.ump)) {
-        deltas <- delta.ump
-      }
-    }
+#' @rdname safeTwoProportionsTest
+#'
+#' @export
+safe.proportion.test <- function(ya, yb, designObj = NULL, pilot = FALSE) {
```

Comment:
je kunt beter geen gebruik maken van functienamen met een punt, want dat is verwarrend over of het een S3 methode is of niet. Ik vind camelCasing veel makkelijker te lezen. Het is wel zo dat er een `prop.test()` bestaat. Je kunt ook een alias maken voor `safe.prop.test()`

Reply:
die punt zit er in om het inderdaad op prop.test te laten lijken. heb jij ook gedaan voor safe.t.test en safe.z.test.